src/main/scala/it/pagopa/interop/agreementprocess/service/impl/PDFCreatorImpl.scala

```diff
@@ -22,22 +21,14 @@ object PDFCreatorImpl extends PDFCreator with PDFManager {
   private[this] val pdfConfigs: PDFConfiguration = PDFConfiguration(resourcesBaseUrl = Some("/agreementTemplate/"))
   private[this] val printedDateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")
 
-  override def create(template: String, eservice: String, producer: String, consumer: String): Future[File] =
+  override def create(template: String, eservice: String, producer: String, consumer: String): Future[Array[Byte]] =
```

Comment:
The `Future.fromTry(Try(x))` is totally useless here, why don't you switch to `Future(x)` ?

Reply:
All this function depends on `getPDF` that return a `ByteArrayOutputStream => Try[Array[Byte]]`, so at the end what we have is a `Try[Array[Byte]]` and no an `Array[Byte]`.
If we want a Future here we must apply `Future.fromTry`, but maybe I'm missing something about your comment.